File: HRMS/HRMS.Web/Modules/ServerTypes/HR/EmployeeSalaryDetailsColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { EmployeeSalaryDetailsRow } from "./EmployeeSalaryDetailsRow";

export interface EmployeeSalaryDetailsColumns {
    ComponentName: Column<EmployeeSalaryDetailsRow>;
    Amount: Column<EmployeeSalaryDetailsRow>;
    ComponentType: Column<EmployeeSalaryDetailsRow>;
}

export class EmployeeSalaryDetailsColumns extends ColumnsBase<EmployeeSalaryDetailsRow> {
    static readonly columnsKey = 'HR.EmployeeSalaryDetails';
    static readonly Fields = fieldsProxy<EmployeeSalaryDetailsColumns>();
}

File: HRMS/HRMS.Web/Modules/HR/EmployeeSalary/EmployeeSalaryDetailsColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace HRMS.HR.Columns;

[ColumnsScript("HR.EmployeeSalaryDetails")]
[BasedOnRow(typeof(EmployeeSalaryDetailsRow), CheckNames = true)]
public class EmployeeSalaryDetailsColumns
{
    [EditLink, Width(200)]
    public string ComponentName { get; set; }
    [Width(120), DisplayFormat("#,##0.00"), AlignRight]
    public decimal Amount { get; set; }
    [Width(100)]
    public string ComponentType { get; set; }
}
